presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkDiskPage.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.List;
+
+public class PrestoSparkDiskPage
+        implements Serializable, PrestoSparkTaskOutput
+{
+    private long broadcastId;
+    private final URI filePathURI;
+    private final long uncompressedSizeInBytes;
+    private final long compressedSizeInBytes;
+    private final List<Long> pageChecksum;
+    private final int positionCount;
+
+    public PrestoSparkDiskPage(long broadcastId,
+            URI filePathURI,
+            long uncompressedSizeInBytes,
+            long compressedSizeInBytes,
+            List<Long> pageChecksum,
+            int positionCount)
+    {
+        this.broadcastId = broadcastId;
+        this.filePathURI = filePathURI;
+        this.uncompressedSizeInBytes = uncompressedSizeInBytes;
+        this.compressedSizeInBytes = compressedSizeInBytes;
+        this.pageChecksum = pageChecksum;
+        this.positionCount = positionCount;
+    }
+
+    public long getUncompressedSizeInBytes()
+    {
+        return uncompressedSizeInBytes;
+    }
+
+    public long getCompressedSizeInBytes()
+    {
+        return compressedSizeInBytes;
+    }
+
+    public URI getFilePathURI()
+    {
+        return filePathURI;
+    }
+
+    public long getBroadcastId()
+    {
+        return broadcastId;
+    }
+
+    public void setBroadcastId(long broadcastId)
```

Comment:
As a rule of thumb we are trying to keep objects immutable in Presto, unless there's a very good reason not to.